src/vs/workbench/contrib/notebook/browser/controller/executeActions.ts

```diff
@@ -522,20 +523,45 @@ registerAction2(class CancelNotebook extends NotebookAction {
 					group: 'navigation/execute',
 					when: ContextKeyExpr.and(
 						NOTEBOOK_HAS_RUNNING_CELL,
-						NOTEBOOK_INTERRUPTIBLE_KERNEL,
+						NOTEBOOK_INTERRUPTIBLE_KERNEL.toNegated(),
 						ContextKeyExpr.equals('config.notebook.globalToolbar', true)
 					)
 				}
 			]
 		});
 	}
+});
 
-	override getEditorContextFromArgsOrActive(accessor: ServicesAccessor, context?: UriComponents): INotebookActionContext | undefined {
-		return getContextFromUri(accessor, context) ?? getContextFromActiveEditor(accessor.get(IEditorService));
-	}
-
-	async runWithContext(accessor: ServicesAccessor, context: INotebookActionContext): Promise<void> {
-		return context.notebookEditor.cancelNotebookCells();
+registerAction2(class InterruptNotebook extends CancelNotebook {
+	constructor() {
+		super({
+			id: INTERRUPT_NOTEBOOK_COMMAND_ID,
+			title: localize('notebookActions.interruptNotebook', "Interrupt"),
```

Comment:
Is this exposed in the Command Palette? If so can you use an `ILocalizedString`?